### ISSUE TITLE
[pub] new rule: license-exists

### ISSUE THREAD

**Issue by pq** [dart-lang/sdk#58244]
If "pubspec.yaml" exists and contains "version" (and not "publish_to:"), then: "LICENSE" should also exist.

/fyi @jonasfj

**Comment by jonasfj** [comment]
Similar to https://github.com/dart-lang/sdk/issues/58245#issuecomment-2517477468 I'd suggest only ignoring it when `publish_to: none` is present (or `version` isn't present).

**Comment by jonasfj** [comment]
@sigurdm do you agree that this would allow us to remove https://github.com/dart-lang/pub/blob/master/lib/src/validator/license.dart

**Comment by sigurdm** [comment]
> @sigurdm do you agree that this would allow us to remove https://github.com/dart-lang/pub/blob/master/lib/src/validator/license.dart

Yes, if it is included by default. (basically is this a lint or warning I guess) If you have to opt-in we need a mechanism to ensure that the check is enabled at publication time.
